operation/fhir-operation-everything/src/main/java/com/ibm/fhir/operation/everything/EverythingOperation.java

```diff
@@ -190,63 +227,82 @@ protected Parameters doInvoke(FHIROperationContext operationContext, Class<? ext
+            if (startOrEndProvided  && !SUPPORT_CLINICAL_DATE_QUERY.contains(compartmentMemberType)) {
+                if (LOG.isLoggable(Level.FINEST)) {
+                    LOG.finest("The request specified a '" + START_QUERY_PARAMETER + "' and/or '" + END_QUERY_PARAMETER + "' query parameter. They are not valid for resource type '" + compartmentMemberType + "', so will be ignored.");
+                }
                 searchParameters = queryParametersWithoutDates;
             }
 
             // Need to construct a new version of the map each time through the loop to edit
-            MultivaluedMap<String, String> tempSearchParameters = 
+            MultivaluedMap<String, String> tempSearchParameters =
                     new MultivaluedHashMap<String, String>(searchParameters);
-            
+
             Bundle results = null;
             int currentResourceCount = 0;
             try {
-                addIncludesSearchParameters(compartmentType, tempSearchParameters);
-                results = resourceHelper.doSearch(compartmentType, PATIENT, logicalId, tempSearchParameters, null, null);
+                // Don't need to add more search parameters if the config section isn't there or is empty
+                if (retrieveAdditionalResources) {
+                    ParametersMap supportedSearchParametersMap = supportedSearchParameters.get(compartmentMemberType);
+                    addIncludesSearchParameters(compartmentMemberType, tempSearchParameters, extraResources, supportedSearchParametersMap);
+                }
+                results = resourceHelper.doSearch(compartmentMemberType, PATIENT, logicalId, tempSearchParameters, null, null);
+                int countBeforeAddingNewResources = allEntries.size();
+                processResults(results, compartmentMemberType, results.getEntry(), allEntries, resourceIds, resourceHelper, extraResources, retrieveAdditionalResources);
                 currentResourceCount = results.getTotal().getValue();
-                totalResourceCount += currentResourceCount;
-                LOG.finest("Got " + compartmentType + " resources " + currentResourceCount + " for a total of " + totalResourceCount);
+                if (LOG.isLoggable(Level.FINEST)) {
+                    LOG.finest("Got " + compartmentMemberType + " resources " + (allEntries.size() - countBeforeAddingNewResources) + " for a total of " + allEntries.size());
+                }
             } catch (Exception e) {
                 FHIROperationException exceptionWithIssue = buildExceptionWithIssue("Error retrieving $everything "
-                        + "resources of type '" + compartmentType + "' for patient " + logicalId, IssueType.EXCEPTION, e);
+                        + "resources of type '" + compartmentMemberType + "' for patient " + logicalId, IssueType.EXCEPTION, e);
```

Comment:
maybe make the same edit as you did above?
```suggestion
                        + "related resources of type '" + compartmentMemberType + "' for patient " + logicalId, IssueType.EXCEPTION, e);
```

Reply:
Done